### ISSUE TITLE
Docker direct build can't COPY or ADD into a volume

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#9105]
Docker build lets you bypass volume logic because it has direct ties to the internals of the daemon.  Direct docker build can't, because it's impossible to start a container and disable a volume (or at least, I'm not aware of any way).  As a result COPY or ADD into a known volume will work within the build, but not be preserved on commit.

We should definitely consider a change to Docker to allow volumes to be disabled at run, which would address this issue for us (we'd start without the volume, then commit it back at the end).  Alternatively, we could provide a volume plugin that makes all volumes act as a no-op (but I don't know that that can change the behavior of the plugin).

@mrunalp is there any movement about letting docker run disable volumes?  Or a no-op volume driver?

**Comment by mrunalp** [comment]
We added a no volume plugin driver for our docker 1.10. 1.10 should be released to Fedora shortly.

**Comment by rhatdan** [comment]
Yes this is the reason we have docker-novolume-plugin.
On my rawhide box.
docker-novolume-plugin-1.11.1-5.git9dea74f.fc25.x86

@runcom PTAL

**Comment by runcom** [comment]
The no volume plugin I've been working on currently error out if you try to run (create + start) a container from an image which has VOLUME defined and the volume isn't overridden at run time. It's not a no op unfortunately... Right now, it's just blocking at start time only but plan is to block directly the  create action so docker root isn't polluted by volumes.

**Comment by smarterclayton** [comment]
The no-op behavior seems strictly more useful, especially for images that
don't provide defaults.  Is it possible via the docker plugin mechanism to
"ignore" the volume request?

**Comment by runcom** [comment]
@smarterclayton No, the plugin just parses the request body of a client call (let's say create) and take decisions based on that. It cannot avoid skipping volumes setup in the daemon.

**Comment by smarterclayton** [comment]
So until we have an alternate container engine implementation we can't
avoid volumes.

**Comment by bparees** [comment]
@smarterclayton "direct docker build"==imagebuilder, right?

**Comment by smarterclayton** [comment]
No, in this case it's running docker build.

**Comment by bparees** [comment]
so what's the feature desire here?  that we be able to mount a PV as part of a "docker build" process, so that certain operations would affect the PV instead of the committed image?
